PowerEditor/src/MISC/Common/Common.cpp

```diff
@@ -1297,6 +1297,15 @@ void trim(generic_string& str)
 	else str.erase(str.begin(), str.end());
 }
 
+bool endsWith(const generic_string& s, const generic_string& suffix)
+{
+#if defined(_MSVC_LANG) && (_MSVC_LANG > 201402L)
+#error Replace this function with basic_string::ends_with
+#endif

```

Comment:
Why this define?

Reply:
Because starting from C++20, this feature is available in standard library https://en.cppreference.com/w/cpp/string/basic_string/ends_with
When the project starts using C++20, this function can be removed.